terraform/modules/prober/prober.tf

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_cloud_run_v2_job" "jvs_prober" {
+
+  project = var.project_id
+
+  name = "jvs-prober"
+
+  location = "us-central1"
+
+  template {
+
+    template {
+
+      containers {
+        image = var.jvs_prober_image
+
+        env {
+          name  = "JVSCTL_SERVER_ADDRESS"
+          value = var.jvs_service_address
+        }
+
+        env {
+          name  = "JVSCTL_JWKS_ENDPOINT"
+          value = var.jvs_jwks_endpoint
+        }
+
+        env {
+          name  = "AUDIENCE"
+          value = var.jvs_audience
+        }
+      }
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [
+      launch_stage,
+    ]
+  }
+}
+
+# This is the service account which will be used by scheduler_jon
+# to trigger jvs-prober cloud run job.
+data "google_compute_default_service_account" "default" {
+  project = var.project_id
+}
```

Comment:
We probably should use a dedicated SA for probing

Reply:
Done, added a service account for prober service as a cloud run invoker.